src/app/shopping-list/shopping-edit/shopping-edit.component.ts

```diff
@@ -1,22 +1,67 @@
-import { Component, OnInit } from '@angular/core';
+import { Subscription } from 'rxjs/Subscription';
+import { Component, OnInit, OnDestroy, ViewChild } from '@angular/core';
 import { ShoppingListService } from '../shopping-list.service';
 import { Ingredient } from 'src/app/shared/model/ingredient.model';
+import { NgForm } from '@angular/forms';
+
 
 
 @Component({
   selector: 'app-shopping-edit',
   templateUrl: './shopping-edit.component.html',
   styleUrls: ['./shopping-edit.component.sass']
 })
-export class ShoppingEditComponent implements OnInit {
+export class ShoppingEditComponent implements OnInit, OnDestroy {
+  @ViewChild('f', null) shoppingListForm: NgForm;
+  startedEditingSubscription: Subscription;
+  editMode = false;
+  editedItemId: number;
```

Comment:
why do you need editedItem and editedItemId since editedItem should contain id?

Reply:
What was changed? Why is it resolved?

Reply:
The editedItem was removed.